client/jetpack-cloud/sections/partner-portal/company-details-form/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { Button, Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { ReactElement, useCallback, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import SelectDropdown from 'calypso/components/select-dropdown';
+import TextPlaceholder from 'calypso/jetpack-cloud/sections/partner-portal/text-placeholder';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, removeNotice, successNotice } from 'calypso/state/notices/actions';
+import useUpdateCompanyDetailsMutation from 'calypso/state/partner-portal/partner/hooks/use-update-company-details';
+import { getCurrentPartner } from 'calypso/state/partner-portal/partner/selectors';
+import { APIError } from 'calypso/state/partner-portal/types';
+import { useCountriesAndStates } from './hooks/use-countries-and-states';
+import { castAsString } from './utils';
+import './style.scss';
+
+export default function CompanyDetailsForm(): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	// The partner has already been fetched through e.g. the requireAccessContext
+	// controller function, so we do not have to do any further checks.
+	const partner = useSelector( getCurrentPartner );
+	const { countryOptions, stateOptions } = useCountriesAndStates();
+	const submitNotificationId = 'company-details-form';
+	const showCountryFields = countryOptions.length > 0;
+
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ name, setName ] = useState( castAsString( partner?.name ) );
+	const [ country, setCountry ] = useState( castAsString( partner?.address.country ) );
```

Comment:
The first time I edit this form, if I don't interact with the Country dropdown the state value will remain "" so submitting the form without doing any changes will result in "Invalid parameter(s): country" because it is an empty string.

Reply:
Ahhh, I just saw there's a "Select country" option so there is an empty option there. My apologies.
In case we want to allow customers to clear their country value we should keep it as-is.
In case we don't want to allow customers to clear their country after they've set it, we should remove the empty option.